src/external/token/ERC20Issuance_Blacklist_v1.sol

```diff
@@ -0,0 +1,212 @@
+
+    /// @notice	Mapping of blacklist manager addresses.
+    mapping(address account => bool isManager) private _isBlacklistManager;
+
+    // -------------------------------------------------------------------------
+    // Modifiers
+
+    /// @notice Modifier to check if the caller is a blacklist manager.
+    modifier onlyBlacklistManager() {
+        if (!_isBlacklistManager[_msgSender()]) {
+            revert ERC20Issuance_Blacklist_NotBlacklistManager();
+        }
+        _;
+    }
+
+    // -------------------------------------------------------------------------
+    // Constructor
+
+    /// @notice	Constructor for ERC20Issuance_Blacklist_v1.
+    /// @param	name_ Token name.
+    /// @param	symbol_ Token symbol.
+    /// @param	decimals_ Token decimals.
+    /// @param	maxSupply_ Max token supply.
+    /// @param	initialAdmin_ Initial admin address.
+    /// @param	initialBlacklistManager_ Initial blacklist manager (typically an
+    ///         EOA).
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        uint8 decimals_,
+        uint maxSupply_,
+        address initialAdmin_,
+        address initialBlacklistManager_
+    ) ERC20Issuance_v1(name_, symbol_, decimals_, maxSupply_, initialAdmin_) {
+        _setBlacklistManager(initialBlacklistManager_, true);
+    }
+
+    // -------------------------------------------------------------------------
+    // View Functions
+
+    /// @inheritdoc	IERC20Issuance_Blacklist_v1
+    function isBlacklisted(address account_) public view returns (bool) {
+        return _blacklist[account_];
+    }
+
+    /// @inheritdoc	IERC20Issuance_Blacklist_v1
+    function isBlacklistManager(address account_) public view returns (bool) {
+        return _isBlacklistManager[account_];
+    }
+
+    // -------------------------------------------------------------------------
+    // External Functions
+
+    /// @inheritdoc IERC20Issuance_Blacklist_v1
+    function addToBlacklist(address account_) public onlyBlacklistManager {
+        if (account_ == address(0)) {
+            revert ERC20Issuance_Blacklist_ZeroAddress();
+        }
+        if (!isBlacklisted(account_)) {
+            _blacklist[account_] = true;
+            emit AddedToBlacklist(account_);
+        }
+    }
+
+    /// @inheritdoc IERC20Issuance_Blacklist_v1
+    function removeFromBlacklist(address account_)
+        public
+        onlyBlacklistManager
+    {
+        if (account_ == address(0)) {
+            revert ERC20Issuance_Blacklist_ZeroAddress();
+        }
+        if (isBlacklisted(account_)) {
+            _blacklist[account_] = false;
+            emit RemovedFromBlacklist(account_);
+        }
+    }
+
+    /// @inheritdoc IERC20Issuance_Blacklist_v1
+    function addToBlacklistBatched(address[] memory accounts_)
+        external
+        onlyBlacklistManager
+    {
+        uint totalAccounts = accounts_.length;
+        if (totalAccounts > BATCH_LIMIT) {
```

Comment:
what is the purpose of this logic? Is there a problem i fthe blacklist manager adds move than BATCH_LIMIT addresses to the blacklist?